lottie/src/main/java/com/airbnb/lottie/network/NetworkFetcher.java

```diff
@@ -85,20 +85,28 @@ private LottieComposition fetchFromCache() {
   @WorkerThread
   private LottieResult fetchFromNetworkInternal() throws IOException {
     L.debug( "Fetching " + url);
-    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
 
+    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
     connection.setRequestMethod("GET");
     connection.connect();
 
     if (connection.getErrorStream() != null || connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+
+      int responseCode = connection.getResponseCode();
       BufferedReader r = new BufferedReader(new InputStreamReader(connection.getErrorStream()));
       StringBuilder error = new StringBuilder();
       String line;
       while ((line = r.readLine()) != null) {
         error.append(line).append('\n');
       }
+
+      // Be sure to clean resources
+      r.close();
```

Comment:
What happens on an exception? Should this be a try with resource instead?

Reply:
That's a good catch. I'm not an expert but it seems the project requirement does not allow me to use this syntax.

Reply:
it's because the minSdk is 16 and that wasn't added until 19

Reply:
I would still wrap this with a try/finally though.